drivers/pds/lib/resiliency_utils.go

```diff
@@ -105,6 +107,22 @@ func InduceFailureAfterWaitingForCondition(deployment *pds.ModelsDeployment, nam
 		if testError != nil {
 			return testError
 		}
+	case KillDeploymentControllerPod:
+		checkTillReplica = CheckTillReplica
```

Comment:
can we directly use CheckTillReplica instead of this assignment?

Reply:
Noted - Will make this change in the next PR (Slightly involved -so will do this in the refinement PR along with using channels for erros)